### DISCUSSION TITLE
What's the difference between field conflicts, field precedences and function Precedence? What kind of scenarios are they applicable to?

### DISCUSSION THREAD

**Discussion by zdaiot** [tree-sitter/tree-sitter#2271] [Q&A]
What's the difference between field conflicts, field precedences and function Precedence? 
What kind of scenarios are they applicable to? 

Can you give me some examples? 

I don't quite understand that I only rely on the text description in https://tree-sitter.github.io/tree-sitter/creating-parsers. 

Thank you very much

**Comment by ahlinc** [comment]
You can try to look on a mature grammar how they are used https://github.com/tree-sitter/tree-sitter-javascript/blob/master/grammar.js
